File: Arrays/ChocolateDistribution/ChocolateDistribution.js
function method1(arr,m){
    arr.sort((a,b)=>a-b);
    let i,count=Number.MAX_VALUE,n=arr.length;
    for(i=0;(i+m-1)<n;i++){
        count = Math.min(count, arr[i+m-1]-arr[i]);
    }
    return count;
}

let arr = [7, 3, 2, 4, 9, 12, 56];
let m = 3;
console.log(method1(arr, m));

File: Arrays/ChocolateDistribution/ChocolateDistribution.go
package main

import (
	"fmt"
	"math"
	"sort"
)

func chocolateDistribution(arr []int, m int) int {
	sort.Ints(arr)
	diff := math.MaxInt
	n := len(arr)
	i := 0
	j := m - 1
	for ; j < n; j++ {
		if arr[j]-arr[i] < diff {
			diff = arr[j] - arr[i]
		}
		i++
	}
	return diff
}

func main() {
	arr := []int{7, 3, 2, 4, 9, 12, 56}
	fmt.Println(chocolateDistribution(arr, 5))
}
